mfkdf2/src/setup/factors/mod.rs

```diff
@@ -1,87 +1,88 @@
-pub enum FactorType {
+pub enum FactorSetupType {
   Password(password::Password),
   HOTP(hotp::HOTP),
-  // Question(question::Question),
-  // UUID(uuid::UUID),
-  // HmacSha1(hmacsha1::HmacSha1),
-  // TOTP(totp::TOTP),
-  // Stack(stack::Stack),
+  Question(question::Question),
+  UUID(uuid::UUID),
+  HmacSha1(hmacsha1::HmacSha1),
+  TOTP(totp::TOTP),
+  OOBA(ooba::Ooba),
+  Passkey(passkey::Passkey),
+  Stack(stack::Stack),
 }
 
-impl FactorType {
-  pub fn inner(&self) -> &dyn FactorTrait {
+impl FactorSetupType {
+  pub fn inner(&self) -> &dyn FactorSetupTrait {
     match self {
-      FactorType::Password(password) => password,
-      FactorType::HOTP(hotp) => hotp,
-      // FactorType::Question(question) => question,
-      // FactorType::UUID(uuid) => uuid,
-      // FactorType::HmacSha1(hmacsha1) => hmacsha1,
-      // FactorType::TOTP(totp) => totp,
-      // FactorType::Stack(stack) => stack,
+      FactorSetupType::Password(password) => password,
+      FactorSetupType::HOTP(hotp) => hotp,
+      FactorSetupType::Question(question) => question,
+      FactorSetupType::UUID(uuid) => uuid,
+      FactorSetupType::HmacSha1(hmacsha1) => hmacsha1,
+      FactorSetupType::TOTP(totp) => totp,
+      FactorSetupType::OOBA(ooba) => ooba,
+      FactorSetupType::Passkey(passkey) => passkey,
+      FactorSetupType::Stack(stack) => stack,
     }
   }
 
-  pub fn inner_mut(&mut self) -> &mut dyn FactorTrait {
+  pub fn inner_mut(&mut self) -> &mut dyn FactorSetupTrait {
     match self {
-      FactorType::Password(password) => password,
-      FactorType::HOTP(hotp) => hotp,
-      // FactorType::Question(question) => question,
-      // FactorType::UUID(uuid) => uuid,
-      // FactorType::HmacSha1(hmacsha1) => hmacsha1,
-      // FactorType::TOTP(totp) => totp,
-      // FactorType::Stack(stack) => stack,
+      FactorSetupType::Password(password) => password,
+      FactorSetupType::HOTP(hotp) => hotp,
+      FactorSetupType::Question(question) => question,
+      FactorSetupType::UUID(uuid) => uuid,
+      FactorSetupType::HmacSha1(hmacsha1) => hmacsha1,
+      FactorSetupType::TOTP(totp) => totp,
+      FactorSetupType::OOBA(ooba) => ooba,
+      FactorSetupType::Passkey(passkey) => passkey,
+      FactorSetupType::Stack(stack) => stack,
     }
   }
 }
 
-impl FactorTrait for FactorType {
+impl FactorSetupTrait for FactorSetupType {
   fn kind(&self) -> String { self.inner().kind() }
 
   fn bytes(&self) -> Vec<u8> { self.inner().bytes() }
 
   fn params_setup(&self, key: [u8; 32]) -> Value { self.inner().params_setup(key) }
 
   fn output_setup(&self, key: [u8; 32]) -> Value { self.inner().output_setup(key) }
-
-  fn params_derive(&self, key: [u8; 32]) -> Value { self.inner().params_derive(key) }
-
-  fn output_derive(&self, key: [u8; 32]) -> Value { self.inner().output_derive(key) }
-
-  fn include_params(&mut self, params: Value) { self.inner_mut().include_params(params) }
 }
 
-pub trait FactorTrait {
+pub trait FactorSetupTrait {
```

Comment:
we may want this:
```suggestion
pub trait FactorSetupTrait: Default {
```

Possibly restrict with `+ Debug` as well.

Reply:
understood. todo added.